mesos-rxjava-test/src/main/java/org/apache/mesos/rx/java/test/TcpSocketProxy.java

```diff
@@ -0,0 +1,187 @@
+package org.apache.mesos.rx.java.test;
+
+import com.google.common.collect.Lists;
+import com.google.common.io.ByteStreams;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A utility class that can used to create a proxy between mesos and your client,
+ * and then close the connection to verify behavior.
+ */
+public final class TcpSocketProxy implements Closeable {
```

Comment:
This was used for manual testing of the exception propagation.

In a future PR there will be some integration tests that will leverage this (wanted to cut down on the PR size though as it was getting pretty big).